packages/iTwinUI-react/src/core/Select/Select.tsx

```diff
@@ -254,25 +344,71 @@ export const Select = <T,>(props: SelectProps<T>): JSX.Element => {
           key: `${option.label}-${index}`,
           isSelected,
           onClick: () => {
-            !option.disabled && onChange?.(option.value);
-            close();
+            if (option.disabled) {
+              return;
+            }
+            if (isSingleOnChange(onChange, multiple)) {
+              onChange?.(option.value);
+              close();
+            } else {
+              onChange?.(option.value, isSelected ? 'removed' : 'added');
+            }
+          },
+          ref: (el: HTMLElement) => {
+            if (isSelected && !multiple) {
+              el?.scrollIntoView();
+            }
           },
-          ref: (el: HTMLElement) => isSelected && el?.scrollIntoView(),
           role: 'option',
           ...option,
           ...menuItem.props,
         });
       });
     },
-    [itemRenderer, onChange, options, value],
+    [itemRenderer, multiple, onChange, options, value],
   );
 
-  const selectedItem = React.useMemo(() => {
+  const selectedItems = React.useMemo(() => {
     if (value == null) {
       return undefined;
     }
-    return options.find((option) => option.value === value);
-  }, [options, value]);
+    return isMultipleEnabled(value, multiple)
+      ? options.filter((option) => value.some((val) => val === option.value))
+      : options.find((option) => option.value === value);
+  }, [multiple, options, value]);
+
+  const tagRenderer = React.useCallback(
+    (item: SelectOption<T>) => {
+      const onCloseKeyDown = (e: React.KeyboardEvent) => {
+        switch (e.key) {
+          case 'Enter':
+          case ' ':
+          case 'Spacebar':
+            focusNextElement(selectRef.current, () =>
```

Comment:
I know that you already know about this but in or der not to forget bringing focus to next tag doesn't work when there overflowed tags.

Reply:
I have fixed it.. 
Need to apply the same fix for `Breadcrumbs` and `ButtonGroup`. 
WIll create an issue.